### PR TITLE
Adding Delivery Employee Table and ability to create new delivery emp…

### PR DESCRIPTION
**Pull request by Brigid481**
Added Delivery Employee Table and ability to add new employees with validation

### CODE REVIEW

File: src/main/java/org/example/Validators/EmployeeValidator.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package org.example.Validators;
+
+import org.example.daos.DeliveryEmployeeDao;
+import org.example.exceptions.Entity;
+import org.example.exceptions.InvalidException;
+import org.example.models.DeliveryEmployeeRequest;
+
+public class EmployeeValidator {
+
+    private DeliveryEmployeeDao dao = new DeliveryEmployeeDao();
```
Comment (bbell-kainos) [inline src/main/java/org/example/Validators/EmployeeValidator.java:10]: Can this be removed if its not being used?

File: src/main/java/org/example/services/DeliveryEmployeeService.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    public int createDeliveryEmployee(
+        final DeliveryEmployeeRequest deliveryEmployee)
+            throws SQLException, InvalidException {
+        employeeValidator.validateEmployee(deliveryEmployee);
+        int id = dao.createDeliveryEmployee(deliveryEmployee);
+
+        if (id == -1) {
+            return -1;
+        }
+        return id;
```
Comment (shaunganley) [inline src/main/java/org/example/services/DeliveryEmployeeService.java:30]: you can do this on one line
```suggestion
        return dao.createDeliveryEmployee(deliveryEmployee);
```

File: src/main/java/org/example/controllers/DeliveryEmployeeController.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.sql.SQLException;
+
+@Api("Delivery Employee API")
+@Path("/api/deliveryemployee")
+public class DeliveryEmployeeController {
+
+    public DeliveryEmployeeService getService() {
```
Comment (bbell-kainos) [inline src/main/java/org/example/controllers/DeliveryEmployeeController.java:20]: Getters/setters not needed here as it is not a model class

File: src/main/resources/db/migration/V1__create_delivery_employee_table.sql [hunk 4]
```diff
@@ -0,0 +1,7 @@
+create table Delivery_Employee (
+    Employee_ID smallint Primary Key AUTO_INCREMENT NOT NULL,
+    Employee_Name varchar(70),
+    Salary decimal(7,2),
+    Employee_BankAccountNumber varchar(8),
+    Employee_NI_Number char(9)
+);
```
Comment (shaunganley) [inline src/main/resources/db/migration/V1__create_delivery_employee_table.sql:7]: could insert some test data here

**Review by shaunganley — APPROVED** [review]

**Review by bbell-kainos — CHANGES_REQUESTED** [review]

**Review by bbell-kainos — APPROVED** [review]

**Pull request merged**